pwiz_tools/BiblioSpec/src/DiaNNSpecLibReader.cpp

```diff
@@ -601,10 +605,18 @@ bool DiaNNSpecLibReader::parseFile()
     auto diannReportFilepath = bal::replace_last_copy(specLibFile, "-lib.tsv.speclib", "-report.tsv");
 
     // special case for FragPipe
-    if (specLibFilePath.filename().string() == "library.tsv.speclib" && bfs::exists(specLibFilePath.parent_path() / "diann-output" / "diann-output.tsv"))
+    if (specLibFilePath.filename().string() == "library.tsv.speclib" || specLibFilePath.filename().string() == "lib.predicted.speclib")
```

Comment:
I wonder if we should simply look for a "diann-output" subdirectory containing either report.tsv or diann-output.tsv. Having these explicit .speclib file names seems less future-proof. Can you think of cases where a .speclib file ends up in a Fragpipe folder with a diann-output subfolder where we would not want to use the contents of that subfolder?

Reply:
It's not a stretch to imagine that a DIA-NN GUI user chooses "diann-output" as their output folder, and report.tsv would be the default report name in that case. The default spectral library name is report-lib.tsv. If I change the default report name to reportfoo, then the default spectral library name becomes reportfoo-lib.tsv. On the other hand, it might be stretching that a GUI user picked the diann-output subdirectory for the report.tsv and the parent directory for the speclib. Unless they plan to re-use that speclib file for multiple reports... I don't have experience using DIA-NN that way, but IIRC you said that's pretty typical?

So I thought "library.tsv.speclib" and  "lib.predicted.speclib" were sufficiently specific to (and hardcoded in) FragPipe because it's not anything like the default. Possibly it would be better to look for the FragPipe log or param file or something to be really sure that it's a FragPipe run, and then skip the report filename check and just do the diann-output check.